doc/embedded/crosscompile_example.md

```diff
@@ -0,0 +1,194 @@
+## Crosscompile mlpack example for an embedded hardware
+
+In this article, we explore how to crosscompile and run an mlpack example code
+on an embedded hardware. In our previous
+[documentations](crosscompile_armv7.cmake), we have explored how to run mlpack
+bindings such as kNN command line program on a Raspberry PI 2. Please refer to that
+article first and follow the first part on how to setup cross-compilation toolchain,
+and then continue with this article. Or refer to this
+[table](supported_boards.md) for a quick start guide.
+
+### Cloning mlpack example respository
+
+mlpack has an [example repository](https://github.com/mlpack/examples) that
+demonstrates a set of examples showing how to use the library source code
+on different dataset and usecases including embedded deployment. This
+tutorial basically explains our necessary CMake configurations
+that are required to integrate with your local CMake to download mlpack
+dependencies and cross compile the entire software.
+
+If you have not used mlpack example repsoitory, I highly recommend to clone
+this repository from the following link:
+
+```sh
+git clone git@github.com:mlpack/examples.git
+```
+
+You can explore this repository and see the available examples. However, in this
+tutorial, we are interested in the `embedded/` directory. In this directory we are providing
+a CMake template project and a RandomForest example to use it on embedded
+hardware. For now, we are exploring the RandomForest example by 
+analysing the CMakeLists.txt.
+
+### Analysing CMakeLists.txt
+
+In this part of code we are defining the project name and including two CMake
+configurations files. The first one is the `Autodowload.cmake` function that is going
+to pull the dependencies from respective locations that we are going to define.
+
+The second configuration file is `ConfigureCrossCompile.cmake`. This will allow
+to include all the necessary configurations to allow using a cross compiler
+toolchain.  Then we are setting the C++ standard to use and define settings
+for `OpenMP` and `std::thread`
```

Comment:
```suggestion
The first part of the code, printed below or [available here](https://github.com/mlpack/examples/blob/master/embedded/crosscompile_random_forest/CMakeLists.txt),
defines the project name and includes two useful CMake configuration files:

 * `Autodownload.cmake`: downloads mlpack's dependencies from defined locations
 * `ConfigureCrossCompile.cmake`: set up CMake configuration for cross-compilation
 
The next steps set the required C++ standard to C++17 (which is the minimum required
version for mlpack), and enable OpenMP for parallelism.
```

Actually the OpenMP support block is not here---and I'm not sure pthreads is actually necessary for this example at all.  So maybe it needs to be removed from the examples repository?